guidelines.md

```diff
@@ -0,0 +1,156 @@
+# Welcome
+Welcome to the rAthena Code Guidelines. 
+This is here to help all developers in creating safe, maintainable, easily readable and understandable code.
+
+# References
+These guidelines are inspired by the CppCoreGuidelines. Some of the examples below maybe taken directly from there.
+If you need more information that may not be listed here, please be sure to check out [CppCoreGuidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md).
+If your questions are still unanswered after consulting this or CppCoreGuidelines, consider consulting the [rAthena forum](https://rathena.org/board/), [rAthena's Discord](https://rathena.org/discord/), or add your concerns directly to a pull request.
+
+# Code Guidelines
+### 1. Don't unnecessarily remove or add blank lines when commiting new code
+
+##### Reason
+
+Commits should be on point, documented and as small as possible.
+
+##### Enforcement
+
+Check for removed or added blank lines on each commit.
+
+### 2. Use std::remove_if, std::remove & ... within std::vector::earse
+
+##### Reason
+
+`std::vector` requires to be shrinked after using `std::remove`. To avoid forgetting about proper shrinking after removing an element the removing should always be bound to a call to `earse`.
+
+##### Note
+
+If using a `std::list` one should rely on `std::list::remove` instead of `std::remove`.
+
+##### Example
+
+    vec.erase(std::remove_if(vec.begin(), vec.end(), [](int a) {return a % 2 == 0;}), vec.end()); // good
+
+##### Enforcement
+
+Check for `std::remove` on datatype `std::vector` without being surrounded by an `erase`.
+
+### 3. Don't add redundant `==` or `!=` to conditions
```

Comment:
I really think that the opposit should be enforced. If you guys want to move to C++, there are going to be a lot of variable with type `auto`. So being explicit in the comparison that you are trying to accomplish would give you a hint in the type of variable that you are dealing with. Even more so, the type could override the `operator bool` and that would complicate things a little bit more...

Reply:
``auto`` is only used if the type is obvious. So the type would be clear in following ``if-statements``.
If the type overrides the ``operator bool`` then its not redundant.